mqtt.go

```diff
@@ -278,7 +278,8 @@ func (mc *mqttConn) runPacketWriter() {
 		if err != nil {
 			logError("[MQTT] failed to send %s packet: %s", p.Type(), err.Error())
 			mc.err <- err
-			break
+			// TODO why we need this break?
```

Comment:
@ethankan do you know why you needed this break?

Because it causes dead lock:
1, because of the error, PktWriter exits. So no more mc.outPacket listening
2, ActiveLoop tries to close gracefully, and it tries to send packet.NewDisconnectPacket() to mc.outPacket. but it's stuck because of #1.
3, SendEvent() try to send an event. But still ActiveLoop is stuck. So the call is also stuck.